core/shortcut_items.ts

```diff
@@ -188,8 +192,27 @@ export function registerPaste() {
       return !workspace.options.readOnly && !Gesture.inProgress();
     },
     callback() {
-      if (!copyData || !copyWorkspace) return false;
-      return !!clipboard.paste(copyData, copyWorkspace);
+      if (!copyData || !copyCoords || !copyWorkspace) return false;
+      const {
+        left: vpLeft,
+        top: vpTop,
+        width: vpWidth,
+        height: vpHeight,
+      } = copyWorkspace.getMetricsManager().getViewMetrics(true);
+      if (
```

Comment:
Add a comment explaining the two cases (when the default location is inside the viewport vs not).